lib/strava/errors/upload_error.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Strava
+  module Errors
+    class UploadError < ::Faraday::ClientError
+      def status
+        response[:status]
+      end
+
+      def headers
+        response[:headers]
+      end
+
+      def message
+        body[:error] || super
+      end
+
+      def error_status
+        body[:status]
+      end
+
+      def upload_failed
+        Strava::Models::Upload.new(body, validate: false)
+      end
```

Comment:
I would love to kick this out, what do you think. not speaking about the validation flag, but handing out an Upload instance as part of the error.

See [here](https://github.com/dblock/strava-ruby-client/pull/23/files#diff-683234911db77b26e29db7e7a2149b1bef3616340a6c74176857ce72f5b6b8d7R5) why I added it in the first place, thanks.

Reply:
Why? It should just be called `upload` and then you get all its properties. I would memoize it too.

```ruby
def upload
  @upload ||= Upload.new(...)
end
```